test/utils/signing.js

```diff
@@ -0,0 +1,17 @@
+const { ethers, network } = require('hardhat');
+const { arrayify, formatBytes32String, defaultAbiCoder, hexConcat, splitSignature, keccak256 } = ethers.utils;
+
+const MEMBERSHIP_APPROVAL = formatBytes32String('MEMBERSHIP_APPROVAL');
+
+const signMembershipApproval = async ({ address, nonce, chainId, kycAuthSigner }) => {
+  const message = defaultAbiCoder.encode(
+    ['bytes32', 'uint256', 'address', 'uint256'],
+    [MEMBERSHIP_APPROVAL, nonce, address, chainId || network.config.chainId],
+  );
+  const hash = keccak256(message);
+  const signature = await kycAuthSigner.signMessage(arrayify(hash));
+  const { r, s, v } = splitSignature(signature);
+  return hexConcat([r, s, v]);
```

Comment:
The splitting is redundant here. The `signature` si exactly what `hexConcat` returns.